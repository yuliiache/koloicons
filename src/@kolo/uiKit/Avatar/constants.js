export const AVATAR_SIZES = Object.freeze({
  WIDTH: 160,
  HEIGHT: 160,
});
