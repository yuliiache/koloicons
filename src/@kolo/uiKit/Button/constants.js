export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export const BUTTON_COLOR = Object.freeze({
  PRIMARY: 'primary',
  ACCENT: 'accent',
  BASIC: 'basic',
});

export const BUTTON_SIZE = Object.freeze({
  SMALL: 'small',
  BIG: 'big',
  FULL_WIDTH: 'full-width',
});

export const BUTTON_TYPE = Object.freeze({
  BUTTON: 'button',
  SUBMIT: 'submit',
  RESET: 'reset',
});
