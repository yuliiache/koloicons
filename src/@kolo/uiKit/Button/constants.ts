export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export enum ButtonColor {
  PRIMARY = 'primary',
  ACCENT = 'accent',
  BASIC = 'basic',
}

export enum ButtonSize {
  SMALL = 'small',
  BIG = 'big',
  FULL_WIDTH = 'full-width',
}

export enum ButtonType {
  BUTTON = 'button',
  SUBMIT = 'submit',
  RESET = 'reset',
}
