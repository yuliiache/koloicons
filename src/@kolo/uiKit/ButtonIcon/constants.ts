export const BUTTON_ICON_TEST_ID = 'buttonIconComponent';

export enum ButtonIconType {
  ROUND = 'round',
  SQUARE = 'square',
}
