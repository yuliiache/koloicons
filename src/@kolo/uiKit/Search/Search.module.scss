@import "@kolo/design";

.searchWrap {
  position: relative;
  display: flex;
  flex-wrap: wrap;
  width: 100%;
  border-left: 1px solid $SeparatorColor;
  border-right: 1px solid $SeparatorColor;

  &.rounded {
    border-radius: 4px;
    border-left: none;
    border-right: none;
  }

  & .searchBtn {
    position: absolute;
    top: 0px;
    left: 16px;
    z-index: 2;

    & > button {
      height: 72px;

      background-color: transparent;
      box-shadow: none;
      cursor: pointer;

      @media (max-width: 767px) {
        height: 48px;
      }

      & .iconSearch path {
        transition: fill $transition-main ease-in-out;
      }

      &:hover,
      &:active {
        .iconSearch path:first-child {
          fill: $LighthIcon;
        }

        .iconSearch path:last-child {
          fill: $IconHoverColor;
        }
      }

      &:focus {
        box-shadow: none;
      }

      &:not(:disabled):active,
      &:not(:disabled):hover {
        background-color: transparent;
        transition: none;
      }
    }
  }

  & .input {
    position: relative;
    height: 72px;
    width: 100%;
    padding: 20px 16px 20px 56px;
    font-size: 0.8rem;
    color: $secFontColor;
    font-family: $mainFontFamily;
    line-height: 1.56;
    background: linear-gradient(180deg, $WhiteBG, $WhiteBG);
    border: none;
    border-bottom: 2px solid rgb(225 224 224 / 10%);
    border-radius: inherit;
    outline: none;
    box-shadow: inset 0 2px 0 0 $WhiteBG, inset 0 24px 35px -4px $ShadowColor, 0 12px 24px 0 $ShadowPanel;

    transition-property: box-shadow, border-color;
    transition-duration: $transition-main;
    transition-timing-function: ease-in-out;

    @media (max-width: 767px) {
      padding-top: 12px;
      padding-bottom: 12px;
      height: 48px;
    }

    &::-webkit-input-placeholder {
      color: $lighthGrey;
      font-family: $mainFontFamily;
      font-size: 0.8rem;
      line-height: 1.5;

      transition: color $transition-main ease-in-out;
    }

    &:hover,
    &:active {
      box-shadow: inset 0 2px 0 0 $WhiteBG, inset 0 24px 35px -4px $ActiveInputColor, 0 12px 24px 0 $ShadowPanel;
    }

    &:focus {
      box-shadow: inset 0 2px 0 0 $WhiteBG, inset 0 24px 35px -4px $ActiveInputColor, 0 12px 24px 0 $ShadowPanel;
      border-bottom: 2px solid $BrandColor;

      &::-webkit-input-placeholder {
        color: $placeholderFocusColor;
      }
    }

    &:hover + .searchBtn .iconSearch path:first-child,
    &:focus + .searchBtn .iconSearch path:first-child {
      fill: $LighthIcon;
    }

    &:hover + .searchBtn .iconSearch path:last-child,
    &:focus + .searchBtn .iconSearch path:last-child {
      fill: $IconHoverColor;
    }
  }

  & .cancelBtn {
    position: absolute;
    top: 16px;
    right: 10px;
    z-index: 2;

    @media (max-width: 767px) {
      top: 4px;
    }

    & > button {
      height: 40px;
      width: 40px;
      background-color: transparent;
      box-shadow: none;
      cursor: pointer;
      fill: rgba(175, 181, 196, 0.8);

      @media (max-width: 767px) {
        padding: 12px;
      }

      &:hover,
      &:active {
        fill: #717886;
      }

      &:focus {
        box-shadow: none;
      }

      &:not(:disabled):active,
      &:not(:disabled):hover {
        background-color: transparent;
      }
    }
  }
}
