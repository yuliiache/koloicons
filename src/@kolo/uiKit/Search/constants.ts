export const SEARCH_INPUT_TEST_ID = 'searchInputComponent';

export const SEARCH_BUTTON_WRAP_TEST_ID = 'searchButtonComponent';

export const CANCEL_BUTTON_WRAP_TEST_ID = 'cancelButtonComponent';