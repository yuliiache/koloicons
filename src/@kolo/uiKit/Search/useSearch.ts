import {KeyCode} from '@kolo/constants/constants';
import {ChangeEvent, useCallback, useEffect, useState} from 'react';

interface SearchHookReturnType {
  inputValue: string;
  onInputChange: (e: ChangeEvent<HTMLInputElement>) => void;
  onClearBtnClick: () => void;
  onSearchHandle: () => void;
}

const useSearch = (onSearch: (value: string) => void): SearchHookReturnType => {
  const [inputValue, setInputValue] = useState('');
  const trimmedValue: string = inputValue.trim();

  const onInputChange = (e: ChangeEvent<HTMLInputElement>): void => {
    setInputValue(e.target.value);
  };

  const onEnterClick: (e: KeyboardEvent) => void = useCallback(
    (e) => {
      if (e.code === KeyCode.ENTER) {
        onSearchHandle();
      }
    },
    [trimmedValue, onSearch]
  );

  useEffect(() => {
    window.addEventListener('keydown', onEnterClick);

    return () => {
      window.removeEventListener('keydown', onEnterClick);
    };
  }, [onEnterClick]);

  const onSearchHandle = () => {
    if (trimmedValue) {
      onSearch(trimmedValue);
    }
  };

  const onClearBtnClick = () => {
    setInputValue('');
  };

  return {
    inputValue,
    onInputChange,
    onClearBtnClick,
    onSearchHandle,
  };
};

export default useSearch;
