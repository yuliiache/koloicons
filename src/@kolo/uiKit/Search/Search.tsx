import classnames from 'classnames';
import {FC} from 'react';

import ButtonIcon from '../ButtonIcon/ButtonIcon';
import {ButtonIconType} from '../ButtonIcon/constants';
import Icon from '../Icons';
import {CANCEL_BUTTON_WRAP_TEST_ID, SEARCH_BUTTON_WRAP_TEST_ID, SEARCH_INPUT_TEST_ID} from './constants';
import styles from './Search.module.scss';
import useSearch from './useSearch';

interface Props {
  placeholder: string;
  onSearch: (value: string) => void;
  isFocused?: boolean;
  isRounded?: boolean;
}

const Search: FC<Props> = ({placeholder, onSearch, isFocused = false, isRounded = true}) => {
  const {inputValue, onInputChange, onSearchHandle, onClearBtnClick} = useSearch(onSearch);
  const searchWrapClasses = classnames(styles.searchWrap, {
    [styles.rounded]: isRounded,
  });

  return (
    <div className={searchWrapClasses}>
      <input
        data-testid={SEARCH_INPUT_TEST_ID}
        className={styles.input}
        type="text"
        id="search-input"
        placeholder={placeholder}
        value={inputValue}
        onChange={onInputChange}
        autoFocus={isFocused}
      />

      <div
        className={styles.searchBtn}
        data-testid={SEARCH_BUTTON_WRAP_TEST_ID}
      >
        <ButtonIcon
          type={ButtonIconType.SQUARE}
          onClick={onSearchHandle}
        >
          <Icon.Search className={styles.iconSearch} />
        </ButtonIcon>
      </div>

      {inputValue && (
        <div
          className={styles.cancelBtn}
          data-testid={CANCEL_BUTTON_WRAP_TEST_ID}
        >
          <ButtonIcon
            id={'search-clear'}
            type={ButtonIconType.SQUARE}
            tooltipLabel="Clear"
            onClick={onClearBtnClick}
          >
            <Icon.Cancel className={styles.iconCancel} />
          </ButtonIcon>
        </div>
      )}
    </div>
  );
};

export default Search;
