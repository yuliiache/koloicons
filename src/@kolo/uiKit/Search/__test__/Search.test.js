import {KeyCode} from '@kolo/constants/constants';
import {cleanup, fireEvent, render, screen} from '@testing-library/react';

import {CANCEL_BUTTON_WRAP_TEST_ID, SEARCH_BUTTON_WRAP_TEST_ID, SEARCH_INPUT_TEST_ID} from '../constants';
import Search from '../Search';

describe('Search component', () => {
  afterEach(cleanup);

  const onSearchMock = jest.fn();
  const mockInputValue = 'a';
  const props = {
    placeholder: 'Search icon',
    onSearch: onSearchMock,
  };
  const renderSearch = (props) => render(<Search {...props} />);

  const testFocusCases = [
    {name: 'without autofocus', props: {...props, isFocused: false}, expectFocus: false},
    {name: 'with autofocus', props: {...props, isFocused: true}, expectFocus: true},
  ];

  it('should render snapshot successfully', () => {
    const {asFragment} = renderSearch(props);

    expect(asFragment()).toMatchSnapshot();
  });

  testFocusCases.forEach(({name, props, expectFocus}) => {
    it(`should render input ${name}`, () => {
      renderSearch(props);

      const input = screen.getByTestId(SEARCH_INPUT_TEST_ID);

      expect(document.activeElement === input).toBe(expectFocus);
    });
  });

  it('should update input value when typing', () => {
    renderSearch(props);

    const input = screen.getByTestId(SEARCH_INPUT_TEST_ID);

    fireEvent.keyDown(input, {key: 'a'});
    fireEvent.change(input, {target: {value: mockInputValue}});

    expect(input).toHaveValue(mockInputValue);
  });

  it('should not render a cancel button', () => {
    renderSearch(props);

    const cancelButton = screen.queryByTestId(CANCEL_BUTTON_WRAP_TEST_ID);

    expect(cancelButton).not.toBeInTheDocument();
  });

  it('should render snapshot with the cancel button successfully', () => {
    const {asFragment} = renderSearch(props);

    const input = screen.getByTestId(SEARCH_INPUT_TEST_ID);

    fireEvent.change(input, {target: {value: mockInputValue}});

    const cancelButton = screen.getByTestId(CANCEL_BUTTON_WRAP_TEST_ID);

    expect(cancelButton).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should call clearing function after the cancel button is clicked', () => {
    const clearInput = jest.fn();
    renderSearch(props);

    const input = screen.getByTestId(SEARCH_INPUT_TEST_ID);

    fireEvent.change(input, {target: {value: mockInputValue}});

    const cancelButtonWrap = screen.getByTestId(CANCEL_BUTTON_WRAP_TEST_ID);
    const cancelButton = cancelButtonWrap.querySelector('button');

    fireEvent.click(cancelButton);

    expect(input).toHaveValue('');
    expect(clearInput).toHaveBeenCalled;
  });

  it('should call onSearch function after the search button is clicked', () => {
    renderSearch(props);
    const input = screen.getByTestId(SEARCH_INPUT_TEST_ID);
    const searchButtonWrap = screen.getByTestId(SEARCH_BUTTON_WRAP_TEST_ID);
    const searchButton = searchButtonWrap.querySelector('button');

    fireEvent.change(input, {target: {value: mockInputValue}});
    fireEvent.click(searchButton);

    expect(onSearchMock).toHaveBeenCalledWith(mockInputValue);
  });

  it('should not call onSearch function when input value is empty and the search button is clicked', () => {
    renderSearch(props);
    const searchButtonWrap = screen.getByTestId(SEARCH_BUTTON_WRAP_TEST_ID);
    const searchButton = searchButtonWrap.querySelector('button');

    fireEvent.click(searchButton);

    expect(onSearchMock).not.toHaveBeenCalled();
  });

  it('should call onSearch function after the "Enter" key is pressed', () => {
    renderSearch(props);
    const input = screen.getByTestId(SEARCH_INPUT_TEST_ID);

    fireEvent.change(input, {target: {value: mockInputValue}});
    fireEvent.keyDown(input, {key: KeyCode.ENTER, code: KeyCode.ENTER});

    expect(onSearchMock).toHaveBeenCalledWith(mockInputValue);
  });

  it('should not call onSearch function when input value is empty and the "Enter" key is pressed', () => {
    renderSearch(props);
    const input = screen.getByTestId(SEARCH_INPUT_TEST_ID);

    fireEvent.keyDown(input, {key: KeyCode.ENTER, code: KeyCode.ENTER});

    expect(onSearchMock).not.toHaveBeenCalled();
  });
});
