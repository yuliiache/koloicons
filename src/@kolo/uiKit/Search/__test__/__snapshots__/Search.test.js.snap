// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should render snapshot successfully 1`] = `
<DocumentFragment>
  <div
    class="searchWrap rounded"
  >
    <input
      class="input"
      data-testid="searchInputComponent"
      id="search-input"
      placeholder="Search icon"
      type="text"
      value=""
    />
    <div
      class="searchBtn"
      data-testid="searchButtonComponent"
    >
      <button
        class="button button-square"
        data-testid="buttonIconComponent"
      >
        <svg
          class="iconSearch"
          data-testid="icon-search"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <path
              d="M11,18 C7.13400675,18 4,14.8659932 4,11 C4,7.13400675 7.13400675,4 11,4 C14.8659932,4 18,7.13400675 18,11 C18,14.8659932 14.8659932,18 11,18 Z"
              fill="#E9EDF7"
            />
            <path
              d="M11,2 C15.9705627,2 20,6.02943725 20,11 C20,13.124779 19.2636898,15.0775836 18.0323074,16.6171757 L21.7071068,20.2928932 C22.0976311,20.6834175 22.0976311,21.3165825 21.7071068,21.7071068 C21.3165825,22.0976311 20.6834175,22.0976311 20.2928932,21.7071068 L20.2928932,21.7071068 L16.6171757,18.0323074 C15.0775836,19.2636898 13.124779,20 11,20 C6.02943725,20 2,15.9705627 2,11 C2,6.02943725 6.02943725,2 11,2 Z M11,4 C7.13400675,4 4,7.13400675 4,11 C4,14.8659932 7.13400675,18 11,18 C12.8904747,18 14.6059153,17.25059 15.8654514,16.0326404 L15.9428932,15.9428932 C15.9713525,15.914434 16.0011003,15.8880487 16.031949,15.8637373 C17.25059,14.6059153 18,12.8904747 18,11 C18,7.13400675 14.8659932,4 11,4 Z"
              fill="#717886"
            />
          </g>
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Search component should render snapshot with the cancel button successfully 1`] = `
<DocumentFragment>
  <div
    class="searchWrap rounded"
  >
    <input
      class="input"
      data-testid="searchInputComponent"
      id="search-input"
      placeholder="Search icon"
      type="text"
      value="a"
    />
    <div
      class="searchBtn"
      data-testid="searchButtonComponent"
    >
      <button
        class="button button-square"
        data-testid="buttonIconComponent"
      >
        <svg
          class="iconSearch"
          data-testid="icon-search"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <path
              d="M11,18 C7.13400675,18 4,14.8659932 4,11 C4,7.13400675 7.13400675,4 11,4 C14.8659932,4 18,7.13400675 18,11 C18,14.8659932 14.8659932,18 11,18 Z"
              fill="#E9EDF7"
            />
            <path
              d="M11,2 C15.9705627,2 20,6.02943725 20,11 C20,13.124779 19.2636898,15.0775836 18.0323074,16.6171757 L21.7071068,20.2928932 C22.0976311,20.6834175 22.0976311,21.3165825 21.7071068,21.7071068 C21.3165825,22.0976311 20.6834175,22.0976311 20.2928932,21.7071068 L20.2928932,21.7071068 L16.6171757,18.0323074 C15.0775836,19.2636898 13.124779,20 11,20 C6.02943725,20 2,15.9705627 2,11 C2,6.02943725 6.02943725,2 11,2 Z M11,4 C7.13400675,4 4,7.13400675 4,11 C4,14.8659932 7.13400675,18 11,18 C12.8904747,18 14.6059153,17.25059 15.8654514,16.0326404 L15.9428932,15.9428932 C15.9713525,15.914434 16.0011003,15.8880487 16.031949,15.8637373 C17.25059,14.6059153 18,12.8904747 18,11 C18,7.13400675 14.8659932,4 11,4 Z"
              fill="#717886"
            />
          </g>
        </svg>
      </button>
    </div>
    <div
      class="cancelBtn"
      data-testid="cancelButtonComponent"
    >
      <button
        class="button button-square"
        data-testid="buttonIconComponent"
      >
        <svg
          class="iconCancel"
          data-testid="icon-cancel"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8,16 C12.418278,16 16,12.418278 16,8 C16,3.581722 12.418278,0 8,0 C3.581722,0 0,3.581722 0,8 C0,12.418278 3.581722,16 8,16 Z M4.20970461,4.38710056 L4.29289322,4.29289322 C4.65337718,3.93240926 5.22060824,3.90467972 5.61289944,4.20970461 L5.70710678,4.29289322 L8,6.585 L10.2928932,4.29289322 C10.6533772,3.93240926 11.2206082,3.90467972 11.6128994,4.20970461 L11.7071068,4.29289322 C12.0675907,4.65337718 12.0953203,5.22060824 11.7902954,5.61289944 L11.7071068,5.70710678 L9.415,8 L11.7071068,10.2928932 C12.0675907,10.6533772 12.0953203,11.2206082 11.7902954,11.6128994 L11.7071068,11.7071068 C11.3466228,12.0675907 10.7793918,12.0953203 10.3871006,11.7902954 L10.2928932,11.7071068 L8,9.415 L5.70710678,11.7071068 C5.34662282,12.0675907 4.77939176,12.0953203 4.38710056,11.7902954 L4.29289322,11.7071068 C3.93240926,11.3466228 3.90467972,10.7793918 4.20970461,10.3871006 L4.29289322,10.2928932 L6.585,8 L4.29289322,5.70710678 C3.93240926,5.34662282 3.90467972,4.77939176 4.20970461,4.38710056 L4.29289322,4.29289322 L4.20970461,4.38710056 Z"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
